### PR TITLE
Update enemy.py

### PR DESCRIPTION
**Pull request by aikios**
i couldn't really understand how you are doing patterns, so i wasn't able to implement those in the class, but if you are willing to explain how you are doing patterns, I could implement that as well - also wasn't sure how to implement damage dealt, so i created a variable that you can update with the interactions with player (again, if you explained how your system works, i could probably do that as well)

I based the phase change system on health thresholds, assuming 4 phases (this can easily be changed to suit your needs)

let me know if you have any questions, and if this project is still being worked on

### CODE REVIEW

**Comment by AdamSpannbauer** [comment]
Hey @deceptikhanqueror thanks for the PR, I like the idea of the `pcb` (and i also like the name). Give me a sec to pull it down and play with it.

And totally fair feedback on the bullet pattern.  We started this project in a very game-jam, just get it going way and the code/direction could be clearer.  I'll try and do some clean up to make this more approachable.

**Comment by aikios** [comment]
cool thanks @AdamSpannbauer , yeah it seems like a very interesting project
hmu if you need anything else, I would be happy to help 

Discord: Khanqueror #7148

File: bullets/enemy.py [hunk 1]
```diff
@@ -1,5 +1,28 @@
 from .actor import Actor
 
-
 class Enemy(Actor):
-    pass
+    def __init__(maxhealth, patterns, pcb1, pcb2, pcb3 )
```
Comment (AdamSpannbauer) [inline bullets/enemy.py]: * the first parameter of a method that uses `self` should be `self`
* `def` lines should end with a `:`

File: bullets/enemy.py [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+        self.health = maxhealth
+        self.maxhealth = maxhealth
+        self.phase = 0
+        self.status = "Alive"
```
Comment (AdamSpannbauer) [inline bullets/enemy.py]: i think i like this being `self.is_alive = True` or potentially using a property/getter like [here in player.py](https://github.com/TriCot-Game-Studio/bullets/blob/main/bullets/player.py#L16-L18).

File: bullets/enemy.py [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+        self.damage = 0
+  ## pcbs are the phase change boundaries (i.e. maxhealth>current_health>pcb1 is 1st (base) phase, pcb1>current_health>pcb2 is 2nd phase, etc. etc. etc.)
+        self.pcb1 = pcb1
+        self.pcb2 = pcb2
+        self.pcb3 = pcb3
+  ## so what i was thinking here is that damage done to the boss is incremented per interval, and then reset every time the boss is updated - this way, everytime boss is updated,
+  ## damage done between updates is incremented to the health and phase, and then the damage is reset to zero for the new interval after having been incremented in the update
+  ## for the previous interval
+    def update(damage)
```
Comment (AdamSpannbauer) [inline bullets/enemy.py]: see comments on the `__init__` method about `self` & `:`

File: bullets/enemy.py [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+        self.health -= damage
+        self.damage = 0
+        if  pcb1 < self.health <= maxhealth:
+            self.phase = 0
+        elif  pcb2 < self.health <= pcb1:
+            self.phase = 1
+        elif  pcb3 < self.health <= pcb2:
+            self.phase = 2
+        elif  0 < self.health <= pcb3:
+            self.phase = 3
```
Comment (AdamSpannbauer) [inline bullets/enemy.py]: `maxhealth` and the `pcb{n}`s need some `self.`s

**Review by AdamSpannbauer — CHANGES_REQUESTED** [review]
Hey @deceptikhanqueror, again, thanks for the PR.  I made some comments on the code.  A lot of what I commented on would be caught by some static code analysis.  This could be done by an IDE (I often use pycharm), or maybe a text editor plugin (i know vscode has pretty good python plugin).  While I'm recommending tools, we've also been using [`black`](https://pypi.org/project/black/) in the project so far to auto-format all of the code.  If you'd prefer not to install, then expect for my to make a formatting commit before merging your PR.

Let me know if you want any help refactoring the code.

**Comment by aikios** [comment]
@AdamSpannbauer made the changes that you requested in new commit 
let me know if I did something wrong

File: bullets/enemy.py [hunk 5]
```diff
@@ -2,4 +2,31 @@
 
 
 class Enemy(Actor):
-    pass
+    def __init__(self, maxhealth, pcb1, pcb2, pcb3):
```
Comment (AdamSpannbauer) [inline bullets/enemy.py]: To inherit all the `Actor` stuff (which admittedly isn't much rn), we need to call the `super()`'s `__init__` method.  You can see this [here in `Player`](https://github.com/TriCot-Game-Studio/bullets/blob/main/bullets/player.py#L5-L6).

Comment (AdamSpannbauer) [reply bullets/enemy.py]: This would also require for `Enemy`'s `__init__()` to get a few more params.

**Review by AdamSpannbauer — CHANGES_REQUESTED** [review]
I think this is the last bit before we'll get this merged this in.

**Comment by aikios** [comment]
OK, I think I did what you wanted, can you confirm that is what you wanted? @AdamSpannbauer

**Review by AdamSpannbauer — APPROVED** [review]

**Pull request merged**